neuralpp/test/quick_tests/symoblic/symbols_test.py

```diff
@@ -0,0 +1,71 @@
+"""
+Tests of features covered in the following tutorials of SymPy.
+https://docs.sympy.org/latest/tutorial/intro.html
+https://docs.sympy.org/latest/tutorial/gotchas.html
+"""
+from sympy import *
+
+
+def test_symbols_basic() -> None:
+    """SymPy supports simple expression conversions. e.g., `x + 2 * y` is the same as `y + x + y`"""
+    x, y = symbols('x y')
+    expr = x + 2 * y
+    assert expr == x + 2 * y
```

Comment:
Interesting! Does that mean the operator == does normalization work every time, or are expressions immediately converted to a normal form as soon as they are created so that equivalent functions are represented identically internally? Does the manual say anything about that? I guess one way to find out is to create two equivalent but distinct expressions and print them to see if their string representation is the same (which would mean they are normalized and the initial form is lost).

Reply:
> Does that mean the operator == does normalization work every time, or are expressions immediately converted to a normal form as soon as they are created so that equivalent functions are represented identically internally?

I remember I read from the doc somewhere that it is the latter, where they gave the example of `x + 1` and `1 + x` are printed in the same way. And for `==`, it is said that "in SymPy, == represents exact structural equality testing."